### PR TITLE
feat: add esmc and esm3 embeddings

### PR DESCRIPTION
**Pull request by jakublala**
## Summary
- add ESM-C and ESM3 embedding-only wrappers in a new shared esm3 runtime family/image
- return residue-aligned embeddings with multichain chain/residue indices and optional logits/hidden states where supported
- document SDK/HF cache/licensing requirements and update registry, image metadata, exports, and contract tests

## Test Plan
- [x] uv run pytest tests/esm3 tests/contracts -q
- [x] uv run python scripts/harness/check_repo.py
- [x] uv run pytest -m "not integration" -q
- [x] uv run --extra dev pre-commit run --all-files

## Notes
- ESM3 generation/folding APIs are intentionally out of scope.
- ESM-C local SDK support is limited to esmc_300m and esmc_600m for the pinned esm SDK.

## Summary by CodeRabbit

* **New Features**
  * Added support for ESM-C and ESM3 embedding models, including new quick-start examples and model listings.
  * Exposed embedding outputs with chain and residue indices for multi-chain inputs.

* **Bug Fixes**
  * Improved handling of shared runtime images so ESM-C and ESM3 use the existing ESMFold2 image.
  * Updated model cache behavior and output padding for more consistent batch results.

* **Documentation**
  * Expanded model, architecture, and image docs with supported backends, aliases, and usage notes.

### CODE REVIEW

**Review by sangioletti — DISMISSED** [review]
Nothing to comment on the implementation, except one caveat (which we might simply correct in a future version, and approve this for now). 

Not sure about ESM-C, since it comes out of the BioHub, but ESM3 had very strict licences for use and in fact it meant that using ESM3 for developing anything commercial in the field of drug design was strictly prohibited. If ESM-C is a better version of ESM3 and has a more permissive licence, I would probably remove ESM3 alltogether?

**Review by sangioletti — APPROVED** [review]
All fine now, let's just keep in mind that Chai also carries a non-commercial licence.
